### PR TITLE
Merge 2.11 to 2.12 [ci: last-only]

### PR DESCRIPTION
**Pull request by lrytz**
```
$ git log --decorate --oneline -1 2.11.x | cat
90215ce (upstream/2.11.x, origin/2.11.x, 2.11.x) Merge pull request #4998 from som-snytt/issue/7898-i
$ git log --decorate --oneline -1 2.12.x | cat
6083875 (HEAD -> 2.12.x, upstream/2.12.x, origin/2.12.x) SI-9767 document and test behaviour of String->integer/float conversions

$ export MB=$(git merge-base 2.12.x 2.11.x)

$ git checkout 2.12.x
$ git checkout -b merge-2.11-to-2.12-june-1
Switched to a new branch 'merge-2.11-to-2.12-june-1'

$ git log --graph --oneline --decorate $MB...2.11.x | cat
*   90215ce (upstream/2.11.x, origin/2.11.x, 2.11.x) Merge pull request #4998 from som-snytt/issue/7898-i
|\
| * b462e5a (origin/pull/4998) SI-7898 Label for parsing -i sources
| * 5450ae6 SI-7898 Report paste errors improvedly
| * 869df33 SI-7898 Quiet REPL at startup
| * 99dad60 SI-7898 Read user input during REPL warmup
* |   971f5ab Merge pull request #5183 from performantdata/bug/SI-9789
|\ \
| * | 100d637 (origin/pull/5183) SI-9789 use quadratic probing in OpenHashMap
|/ /
* |   139f6bf Merge pull request #4959 from rjolly/scripting15
|\ \
| * | 206a657 (origin/pull/4959) Jarlist scala-library in build.sbt
| * | a412ea0 Use jarlister in build
| |/
* |   cba585d Merge pull request #5061 from performantdata/benchmark-framework
|\ \
| * | 486821b (origin/pull/5061) Enable full compiler optimizations in JMH benchmarking.
| * | 00cbba1 Address JMH benchmark reviewer's issues.
| * | cd7be12 Improve the OpenHashMapBenchmark run times.
| * | 2e8fb12 Add a JMH runner class to the library benchmark framework.
| * | b88933e Benchmark the OpenHashMap memory usage. Also add sbteclipse to the benchmark project.
| * | e1b58cc Add get() tests to OpenHashMap, reduce timing artifacts.
| * | 303130b Add a reference to Doug Lea's benchmarks.
| * | 56c5d92 Add JMH to the benchmark framework. Add an example benchmark for OpenHashMap.
|  /
* |   9a703b4 Merge pull request #5169 from som-snytt/issue/4625
|\ \
| * | 7f514bb (origin/pull/5169) SI-4625 Warn on first non-toplevel only
| * | e753135 SI-4625 Warn when discarding script object
| * | ee365cc SI-4625 Permit arbitrary top-level in script
| * | 5bcefbe SI-4625 App is a thing
| * | 73ca44b SI-4625 Recognize App in script
|  /
* |   f640fa0 Merge pull request #5167 from martijnhoekstra/SI-9766
|\ \
| * | c89c365 (origin/pull/5167) SI-9766 - allow ++ on empty ConcatIterator
|  /
* | 4196569 SI-9688 Make merge in immutable HashMap1 work with null kv.
|/
*   0d0671a Merge pull request #5178 from lrytz/readme211
|\
| * e483e49 (origin/pull/5178) [backport] identical readme in 2.11.x as in 2.12.x
|/
*   90706b0 (scala1/2.11.x) Merge pull request #5172 from lrytz/move-test
|\
| * c2f67d2 (origin/pull/5172) Move t8449 to correct place
* 443fa04 Remove default value for sourcepath in scalac (ant version). (#5166)


$ git merge 90706b0
Auto-merging src/compiler/scala/tools/ant/Scalac.scala
Merge made by the 'recursive' strategy.


$ git log --decorate --oneline -1 | cat
7f090f1 (HEAD -> merge-2.11-to-2.12-june-1) Merge commit '90706b0' into merge-2.11-to-2.12-june-1
$ git merge 0d0671a -s ours
Merge made by the 'ours' strategy.
$ git diff 7f090f1 | cat
$


$ git merge 4196569
Auto-merging src/library/scala/collection/immutable/HashMap.scala
Merge made by the 'recursive' strategy.


$ git log --decorate --oneline -1 | cat
460fa9d (HEAD -> merge-2.11-to-2.12-june-1) Merge commit '4196569' into merge-2.11-to-2.12-june-1
$ git merge f640fa0 -s ours
Merge made by the 'ours' strategy.
$ git diff 460fa9d | cat
$


$ git merge cba585d
Auto-merging src/reflect/scala/reflect/internal/StdNames.scala
Auto-merging src/compiler/scala/tools/nsc/ast/parser/Parsers.scala
Merge made by the 'recursive' strategy.


$ git merge 139f6bf
CONFLICT (modify/delete): test/files/run/t7933.scala deleted in HEAD and modified in 139f6bf. Version 139f6bf of test/files/run/t7933.scala left in tree.
CONFLICT (modify/delete): test/files/run/t7843-jsr223-service.scala deleted in HEAD and modified in 139f6bf. Version 139f6bf of test/files/run/t7843-jsr223-service.scala left in tree.
  - these two test were removed in https://github.com/scala/scala/pull/4819, so remove them in the merge
Auto-merging project/Osgi.scala
CONFLICT (content): Merge conflict in project/Osgi.scala
  - conflict in "bundle <<=", the code is slightly different in 2.12.x
Auto-merging build.xml
CONFLICT (content): Merge conflict in build.xml
  - conflict in target "pack.lib", as it does not depend on "forkjoin.done" in 2.12.x
Auto-merging build.sbt
Automatic merge failed; fix conflicts and then commit the result.
$ git rm test/files/run/t7843-jsr223-service.scala test/files/run/t7933.scala
test/files/run/t7843-jsr223-service.scala: needs merge
test/files/run/t7933.scala: needs merge
rm 'test/files/run/t7843-jsr223-service.scala'
rm 'test/files/run/t7933.scala'
$ git add -A

  - a manual fix to Osgi.scala was required: needed to add an `asScala` call to `jar.getResources`,
    as the file imports JavaConverters._ instead of JavaConversions._ in 2.12.x

$ git commit
[merge-2.11-to-2.12-june-1 8384ebd] Merge commit '139f6bf' into merge-2.11-to-2.12-june-1


$ git merge 90215ce
Auto-merging src/repl/scala/tools/nsc/interpreter/IMain.scala
Auto-merging src/repl/scala/tools/nsc/interpreter/ILoop.scala
CONFLICT (content): Merge conflict in src/repl/scala/tools/nsc/interpreter/ILoop.scala
  - conflict just because 2.12.x uses `Future` instead of `future`
Auto-merging src/library/scala/collection/mutable/OpenHashMap.scala
CONFLICT (content): Merge conflict in src/library/scala/collection/mutable/OpenHashMap.scala
  - change in findIndex conflicts with https://github.com/scala/scala/pull/5124/files
Auto-merging src/compiler/scala/tools/nsc/GenericRunnerSettings.scala
Automatic merge failed; fix conflicts and then commit the result.
$ git add -A
$ git commit
[merge-2.11-to-2.12-june-1 6c4414e] Merge commit '90215ce' into merge-2.11-to-2.12-june-1
```

### CODE REVIEW

**Comment by lrytz** [comment]
review by @szeiger

File: test/benchmarks/build.sbt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+scalaHome := Some(file("../../build/pack"))
+scalaVersion := "2.11.8"
```
Comment (szeiger) [inline test/benchmarks/build.sbt]: I suppose this should be "2.12.0" (not sure whether it matters if `scalaHome` is set)

Comment (lrytz) [inline test/benchmarks/build.sbt]: ok, done

**Comment by szeiger** [comment]
LGTM

**Pull request merged**